SML + Bluetooth results in build failure

### PROBLEM DESCRIPTION
Building Tasmota for ESP32C3 with BLE and SML results in the following build failure:

```
In file included from /root/.platformio/packages/framework-arduinoespressif32/tools/esp32-arduino-libs/esp32c3/include/bt/host/nimble/nimble/porting/nimble/include/os/os.h:49,
                 from /root/.platformio/packages/framework-arduinoespressif32/tools/esp32-arduino-libs/esp32c3/include/bt/host/nimble/nimble/nimble/include/nimble/ble.h:26,
                 from lib/libesp32_div/esp-nimble-cpp/src/NimBLEAddress.h:21,
                 from lib/libesp32_div/esp-nimble-cpp/src/NimBLEAdvertisedDevice.h:20,
                 from lib/libesp32_div/esp-nimble-cpp/src/NimBLEScan.h:20,
                 from lib/libesp32_div/esp-nimble-cpp/src/NimBLEDevice.h:22,
                 from /tmp/Tasmota/tasmota/tasmota_xdrv_driver/xdrv_79_esp32_ble.ino:155:
/root/.platformio/packages/framework-arduinoespressif32/tools/esp32-arduino-libs/esp32c3/include/bt/host/nimble/nimble/porting/nimble/include/os/endian.h:32:6: error: expected ')' before '(' token
   32 |      ((((x) & 0xff00000000000000ull) >> 56) |       \
      |      ^
/root/.platformio/packages/framework-arduinoespressif32/tools/esp32-arduino-libs/esp32c3/include/bt/host/nimble/nimble/porting/nimble/include/os/endian.h:133:21: note: in expansion of macro 'os_bswap_64'
  133 | #define ntohll(x)   os_bswap_64(x)
      |                     ^~~~~~~~~~~
lib/lib_div/ams/ntohll.h:6:10: note: in expansion of macro 'ntohll'
    6 | uint64_t ntohll(uint64_t x);
      |          ^~~~~~
/root/.platformio/packages/framework-arduinoespressif32/tools/esp32-arduino-libs/esp32c3/include/bt/host/nimble/nimble/porting/nimble/include/os/endian.h:31:26: note: to match this '('
   31 | #define os_bswap_64(x)   ((uint64_t)                \
      |                          ^
/root/.platformio/packages/framework-arduinoespressif32/tools/esp32-arduino-libs/esp32c3/include/bt/host/nimble/nimble/porting/nimble/include/os/endian.h:133:21: note: in expansion of macro 'os_bswap_64'
  133 | #define ntohll(x)   os_bswap_64(x)
      |                     ^~~~~~~~~~~
lib/lib_div/ams/ntohll.h:6:10: note: in expansion of macro 'ntohll'
    6 | uint64_t ntohll(uint64_t x);
      |          ^~~~~~
```

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Problem is not scripter related, in this case open a discussion and tag  gemu2015
- [x] Device used (e.g., Sonoff Basic): ESP32C3
- [x] Tasmota binary firmware version number used: 14.0, 14.6
  - [ ] Pre-compiled
  - [x] Self-compiled
- [x] Flashing tools used: n/a (compiler error)
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`: n/a
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`: n/a
- [x] Provide the output of this command: `Status 0`: n/a
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log: n/a

### TO REPRODUCE
1. Enable Bluetooth, which pulls in esp32c3's nimble library, which defines `ntohll` as a macro
2. Enable USE_SML_M, which in turn enables `lib/lib_div/ams` from Tasmota's code, which defines `ntohll` as a function
3. Build Tasmota and get the above compile error

(I used TasmoCompile, but that shouldn't really matter as far as I can see)

### EXPECTED BEHAVIOUR
A Tasmota binary with BLE and SML support :)

### SCREENSHOTS
Shouldn't be needed.

### ADDITIONAL CONTEXT
n/a

**(Please, remember to close the issue when the problem has been addressed)**

--- a/lib/lib_div/ams/han_Parser.h
+++ b/lib/lib_div/ams/han_Parser.h
@@ -6,7 +6,7 @@
 #include "DataParsers.h"
 #include "DataParser.h"
 #include "Cosem.h"
-#include "ntohll.h"
+#include "ntohll_ams.h"
 
 #define BUF_SIZE_HAN (1280)
 

--- a/lib/lib_div/ams/ntohll_ams.cpp
+++ b/lib/lib_div/ams/ntohll_ams.cpp
@@ -1,5 +1,5 @@
-#include "ntohll.h"
+#include "ntohll_ams.h"
 
-uint64_t ntohll(uint64_t x) {
+uint64_t ntohll_ams(uint64_t x) {
     return (((uint64_t)ntohl((uint32_t)x)) << 32) + ntohl(x >> 32);
 }

--- a/lib/lib_div/ams/ntohll_ams.h
+++ b/lib/lib_div/ams/ntohll_ams.h
@@ -3,6 +3,6 @@
 
 #include "lwip/def.h"
 
-uint64_t ntohll(uint64_t x);
+uint64_t ntohll_ams(uint64_t x);
 
 #endif

--- a/tasmota/tasmota_xsns_sensor/xsns_53_sml.ino
+++ b/tasmota/tasmota_xsns_sensor/xsns_53_sml.ino
@@ -113,6 +113,15 @@
 #define USE_SML_MEDIAN_FILTER
 #endif
 
+// on ESP32 binary SML CRC check compile enabled by default
+// adds 17936 bytes due to large crc tables
+#ifdef ESP32
+#ifndef NO_USE_SML_CRC
+#undef USE_SML_CRC
+#define USE_SML_CRC
+#endif
+#endif
+
 
 #ifdef USE_SML_DECRYPT
 #include "han_Parser.h"
@@ -248,8 +257,8 @@ private:
   HardwareSerial *hws;
 };
 
-
-void IRAM_ATTR sml_callRxRead(void *self) { ((SML_ESP32_SERIAL*)self)->rxRead(); };
+void IRAM_ATTR sml_callRxRead(void *self);
+void sml_callRxRead(void *self) { ((SML_ESP32_SERIAL*)self)->rxRead(); };
 
 SML_ESP32_SERIAL::SML_ESP32_SERIAL(uint32_t index) {
   uart_index = index;
@@ -373,7 +382,7 @@ void SML_ESP32_SERIAL::updateBaudRate(uint32_t baud) {
 }
 
 // no wait mode only 8N1  (or 7X1, obis only, ignoring parity)
-void IRAM_ATTR SML_ESP32_SERIAL::rxRead(void) {
+void SML_ESP32_SERIAL::rxRead(void) {
   uint32_t diff;
   uint32_t level;
 
@@ -1722,6 +1731,10 @@ void sml_shift_in(uint32_t meters, uint32_t shard) {
           if (mp->spos == 6 + tlen) {
             mp->spos = 0;
             memmove(&mp->sbuff[0], &mp->sbuff[6], mp->sbsiz - 6);
+#ifdef MODBUS_DEBUG
+            AddLog(LOG_LEVEL_INFO, PSTR("meter %d, receive index >> %d"), meters, mp->index);
+            Hexdump(mp->sbuff, 10);
+#endif
             SML_Decode(meters);
             if (mp->client) {
               mp->client->flush();
@@ -1737,7 +1750,7 @@ void sml_shift_in(uint32_t meters, uint32_t shard) {
         if (mlen > mp->sbsiz) mlen = mp->sbsiz;
         if (mp->spos >= mlen) {
 #ifdef MODBUS_DEBUG
-          AddLog(LOG_LEVEL_INFO, PSTR("receive index >> %d"), mp->index);
+          AddLog(LOG_LEVEL_INFO, PSTR("meter %d, receive index >> %d"), meters, mp->index);
           Hexdump(mp->sbuff, 10);
 #endif
           SML_Decode(meters);
@@ -1917,11 +1930,11 @@ double sml_get_obis_value(uint8_t *data) {
 				break;
 		}
 		case CosemTypeLong64Signed: {
-				out = ntohll(item->l64s.data);
+				out = ntohll_ams(item->l64s.data);
 				break;
 		}
 		case CosemTypeLong64Unsigned: {
-				out = ntohll(item->l64u.data);
+				out = ntohll_ams(item->l64u.data);
 				break;
 		}
 	}
@@ -2312,6 +2325,13 @@ void SML_Decode(uint8_t index) {
               mp += 2;
             } else if (!strncmp_P(mp, PSTR("ffffffff"), 8)) {
               uint32_t val = (cp[0]<<24) | (cp[1]<<16) | (cp[2]<<8) | (cp[3]<<0);
+              float *fp = (float*)&val;
+              ebus_dval = *fp;
+              mbus_dval = *fp;
+              mp += 8;
+              cp += 4;
+            } else if (!strncmp_P(mp, PSTR("FFFFFFFF"), 8)) {
+              uint32_t val = (cp[0]<<0) | (cp[1]<<0) | (cp[2]<<16) | (cp[3]<<24);
               float *fp = (float*)&val;
               ebus_dval = *fp;
               mbus_dval = *fp;
@@ -4662,7 +4682,7 @@ void SML_Send_Seq(uint32_t meter, char *seq) {
 #ifdef MODBUS_DEBUG
   uint8_t type = mp->type;
   if (!sml_globs.dump2log && (type == 'm' || type == 'M' || type == 'k')) {
-    AddLog(LOG_LEVEL_INFO, PSTR("transmit index >> %d"),sml_globs.mp[meter].index);
+    AddLog(LOG_LEVEL_INFO, PSTR("meter %d, transmit index >> %d"), meter, sml_globs.mp[meter].index);
     Hexdump(sbuff, slen);
   }
 #endif

